src/main/java/de/hysky/skyblocker/skyblock/dwarven/CorpseFinder.java

```diff
@@ -0,0 +1,269 @@
+        MinecraftClient.getInstance().player.sendMessage(
+            Constants.PREFIX.get()
+                .append("Found a ")
+                .append(Text.literal(corpse.name + " Corpse")
+                    .withColor(corpse.color.getColorValue()))
+                .append(" at " + corpse.entity.getBlockPos().up(0).toShortString() + "!")
+                .styled(style -> style.withClickEvent(new ClickEvent(ClickEvent.Action.RUN_COMMAND, "/skyblocker corpseHelper shareLocation " + PosUtils.toSpaceSeparatedString(corpse.waypoint.pos) + " " + corpse.name))
+                    .withHoverEvent(new HoverEvent(HoverEvent.Action.SHOW_TEXT, Text.literal("Click to share the location in chat!").formatted(Formatting.GREEN)))));
+    }
+
+    static String getType(ArmorStandEntity entity) {
+        for (ItemStack stack : entity.getArmorItems()) {
+            String itemId = ItemUtils.getItemId(stack);
+            switch (itemId) {
+                case ("LAPIS_ARMOR_HELMET"):
+                    return "LAPIS";
+                case ("ARMOR_OF_YOG_HELMET"):
+                    return "UMBER";
+                case ("MINERAL_HELMET"):
+                    return "TUNGSTEN";
+                case ("VANGUARD_HELMET"):
+                    return "VANGUARD";
+            }
+        }
+        return "UNKNOWN";
+    }
+
+    private static Formatting getColor(ArmorStandEntity entity) {
+        for (ItemStack stack : entity.getArmorItems()) {
+            String itemId = ItemUtils.getItemId(stack);
+            if (ITEM_IDS.contains(itemId)) {
+                switch (itemId) {
+                    case ("LAPIS_ARMOR_HELMET"), ("VANGUARD_HELMET"): return Formatting.BLUE; // dark blue looks bad and those two never exist in same shaft
+                    case ("ARMOR_OF_YOG_HELMET"): return Formatting.RED;
+                    case ("MINERAL_HELMET"): return Formatting.GRAY;
+                }}}
+        LOGGER.warn(PREFIX + "Couldn't match a color! Something probably went very wrong!");
+        return Formatting.YELLOW;
+    }
+
+    @SuppressWarnings("DataFlowIssue")
+    private static float[] getColors(Formatting color) {
+        return ColorUtils.getFloatComponents(color.getColorValue());
+    }
+
+    private static void parseCords(Text text) {
+        Pattern cordsPattern = Pattern.compile("x: (?<x>-?\\d+), y: (?<y>-?\\d+), z: (?<z>-?\\d+)(?:.+)?");
+        String message = text.getString();
+        Matcher matcher = cordsPattern.matcher(message);
+        if (matcher.find()) {
+            int x = Integer.parseInt(matcher.group("x"));
+            int y = Integer.parseInt(matcher.group("y"));
+            int z = Integer.parseInt(matcher.group("z"));
+            LOGGER.info(PREFIX + "Parsed message! X:{}, Y:{}, Z:{}", x, y, z);
+            boolean foundCorpse = false;
+            BlockPos parsedPos = new BlockPos(x-1, y-1, z-1);   // skyhanni cords format difference is -1, -1, -1
```

Comment:
is skyhanni the only other mod to send cwords? surely this would mess up detecting cwords from this mods messages.  Would it be better just to see if they are within 2 blocks distance to a location instead of relying on this conversion?
```java
lockPos parsedPos = new BlockPos(x, y, z);
...
...
if(corpse.waypoint.pos.getSquaredDistance(parsedPos) < 4) {
```

Reply:
If you're talking in terms of the temporary waypoint that gets added when someone sends a location, the way I wrote it wouldn't cause an issue. it would just add the corpse location to a list so it doesn't get shared since someone already shared it's location, parsing the waypoint was using skyhanni patcher coords waypoint feature

Reply:
tho reading it again, you'd probably want to use a range of blocks rather than checking corpse location, when I wrote it on SH I didn't see any other mod that had it like this so I just sent the corpse location, but other mods might share the player location.
and also some people just don't realise features like this exist so they just `/patcher sendcoords`